File: JAVA/02_Recursion/Lecture09/pdi.java
package Lecture09;

public class pdi {

	public static void main(String[] args) {
		pdi(6);
	}
	public static void pdi(int n) {
		if(n == 0) {
			return;
		}
		System.out.print(n + " ");
		pdi(n - 1);
		System.out.print(n + " ");
	}

}

File: C++/DS1/Lec9/pdi.cpp
#include <iostream>
using namespace std;
void pdi(int n){
    if(n == 0){
        return;
    }
    cout << n << " ";
    pdi(n - 1);
    cout << n << " ";
}
int main(){
    pdi(5);
    cout << endl;
    return 0;
}